[Feat/application] application serializer 작성

Kang9805 opened the pull request:
## ✅ PR 요약
- application serializers.py 작성완료

## 📄 상세 내용
- [x] Application 관련 시리얼라이저 완료
- [x] 코드 재사용성 향상을 위한 Mixin을 도입하여 중복코드 제거 
- [x] Mypy 타입 안전성 강화

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Kang9805 committed:
✨feat : application_serializers.py 작성

application serializer 작성
AppliedAtSerializerMixin 활용

2joonkim commented on apps/application/serializers/application_serializers.py:
```diff
@@ -0,0 +1,134 @@
+            "has_study_experience",
+            "study_experience",
+        ]
+        read_only_fields: list[str] = []
```
요거는 read_only_fields가 없는거 같은데 따로 없애셔도 무방합니다!

2joonkim commented on apps/application/serializers/application_serializers.py:
```diff
@@ -0,0 +1,134 @@
+
+    def validate(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        has_exp = data.get("has_study_experience")
+        study_exp_content = data.get("study_experience", "").strip()
```
지금 현재 이 코드 보시면 
```python
data.get("study_experience", "")
```
는 키가 없을 때만 ""를 반환합니다
키가 있는데 값이 None이면 그대로 None이 반환되어 .strip()에서 AttributeError터질거에요

이쪽 부분을 바꿔보자면
```python
study_exp_content = (data.get("study_experience") or "").strip()
```
이렇게 된다면 의도하신대로 None값이 들어오더라도 ""로 처리가 되겠죵

Kang9805 replied:
오호 확인했습니다.

2joonkim commented on apps/application/serializers/application_serializers.py:
```diff
@@ -0,0 +1,134 @@
+        if has_exp and not study_exp_content:
+            raise ValidationError(
+                {"study_experience": "스터디 경험 유무를 '있음'으로 선택한 경우 상세 내용을 기재해야 합니다."}
+            )
+        return data
```
요 부분은 ValidationError 잘 이용하셨는데
스터디 경험유무를 '없음'으로 선택했을 때의 상세내용 기재를 못한다는 ValidationError도 필요하지 않을까요?
현재 위 코드만 본다면 빈값으로 들어와도 예외처리가 안되어있습니다.

의도적으로 처리를 하신건지 예외처리를 놓치신건지 확인 한번만 부탁드릴게용!

Kang9805 replied:
확인했습니다.

Kang9805 committed:
💡 chore : application_serializers 수정(타입 안정성 및 명확성 강화)

1. None 값에 대한 안전한 .strip() 처리 로직으로 AttributeError 방지
2. Read-only 필드를 명시적으로 선언하여 가독성 및 유지보수성 향상
3. Recruiter/Applicant 역할 기반으로 클래스명 변경
